lib/block-supports/layout.php

```diff
@@ -371,16 +371,24 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		return (string) $content;
 	}
 
-	$block_gap              = gutenberg_get_global_settings( array( 'spacing', 'blockGap' ) );
-	$global_layout_settings = gutenberg_get_global_settings( array( 'layout' ) );
-	$has_block_gap_support  = isset( $block_gap ) ? null !== $block_gap : false;
-	$default_block_layout   = _wp_array_get( $block_type->supports, array( '__experimentalLayout', 'default' ), array() );
-	$used_layout            = isset( $block['attrs']['layout'] ) ? $block['attrs']['layout'] : $default_block_layout;
-
-	if ( isset( $used_layout['inherit'] ) && $used_layout['inherit'] ) {
-		if ( ! $global_layout_settings ) {
-			return $block_content;
-		}
+	static $static_information_computed   = false;
+	static $has_block_gap_support         = false;
+	static $global_layout_settings        = null;
+	static $root_padding_aware_alignments = false;
+	if ( ! $static_information_computed ) {
+		$global_settings               = gutenberg_get_global_settings();
+		$block_gap                     = _wp_array_get( $global_settings, array( 'spacing', 'blockGap' ), $global_settings );
```

Comment:
The fallback / default value of `array( 'spacing', 'blockGap' )` should be null:

```suggestion
		$block_gap                     = _wp_array_get( $global_settings, array( 'spacing', 'blockGap' ), null );
```